packages/ui/SurveysList/actions.ts

```diff
@@ -29,251 +26,54 @@ export const getSurveyAction = async (surveyId: string) => {
-            actionClassId: actionClassId,
-          })),
-        },
-        attributeFilters: {
-          create: existingSurvey.attributeFilters.map((attributeFilter, idx) => ({
-            attributeClassId: targetEnvironmentAttributeFilters[idx],
-            condition: attributeFilter.condition,
-            value: attributeFilter.value,
-          })),
-        },
-        environment: {
-          connect: {
-            id: parsedInput.targetEnvironmentId,
-          },
-        },
-        creator: {
-          connect: {
-            id: ctx.user.id,
-          },
-        },
-        isVerifyEmailEnabled: existingSurvey.isVerifyEmailEnabled,
-        surveyClosedMessage: existingSurvey.surveyClosedMessage ?? prismaClient.JsonNull,
-        singleUse: existingSurvey.singleUse ?? prismaClient.JsonNull,
-        productOverwrites: existingSurvey.productOverwrites ?? prismaClient.JsonNull,
-        styling: existingSurvey.styling ?? prismaClient.JsonNull,
-        segment: undefined,
-      },
-    });
-
-    // if the existing survey has an inline segment, we copy the filters and create a new inline segment and connect it to the new survey
-    if (existingSurvey.segment) {
-      if (existingSurvey.segment.isPrivate) {
-        const newInlineSegment = await createSegment({
-          environmentId: parsedInput.environmentId,
-          title: `${newSurvey.id}`,
-          isPrivate: true,
-          surveyId: newSurvey.id,
-          filters: existingSurvey.segment.filters,
-        });
-
-        await prisma.survey.update({
-          where: {
-            id: newSurvey.id,
-          },
-          data: {
-            segment: {
-              connect: {
-                id: newInlineSegment.id,
-              },
-            },
-          },
-        });
+export const getProductsByEnvironmentIdAction = async (environmentId: string) => {
+  const session = await getServerSession(authOptions);
+  if (!session) throw new AuthorizationError("Not authorized");
 
-        segmentCache.revalidate({
-          id: newInlineSegment.id,
-          environmentId: newSurvey.environmentId,
-        });
-      } else {
-        await prisma.survey.update({
-          where: {
-            id: newSurvey.id,
-          },
-          data: {
-            segment: {
-              connect: {
-                id: existingSurvey.segment.id,
-              },
-            },
-          },
-        });
+  const organization = await getOrganizationByEnvironmentId(environmentId);
 
-        segmentCache.revalidate({
-          id: existingSurvey.segment.id,
-          environmentId: newSurvey.environmentId,
-        });
-      }
-    }
+  if (!organization) {
+    throw new Error("No organization found");
+  }
 
-    surveyCache.revalidate({
-      id: newSurvey.id,
-      environmentId: parsedInput.targetEnvironmentId,
-    });
-    return newSurvey;
-  });
+  const products = await getProducts(organization.id);
```

Comment:
!!SECURITY ISSUE: As far as I can see we don't check if the user has access to the environment